lib/PuppeteerSharp/WebSocketFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PuppeteerSharp
+{
+    /// <summary>
+    /// Factory method for connected web sockets.
+    /// </summary>
+    /// <param name="uri">The uri to connect to.</param>
+    /// <param name="options">Web socket creation options.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    /// <returns>A <see cref="WebSocket"/> that is connected to the given <paramref name="uri"/>.</returns>
+    public delegate Task<WebSocket> WebSocketFactory(Uri uri, IWebSocketOptions options, CancellationToken cancellationToken);
```

Comment:
I'd just declare that as an Action instead of creating a delegate.

Reply:
That would become a `Func<Uri, IWebSocketOptions, CancellationToken, Task<WebSocket>>` which is significantly less self-documenting than the proposed delegate.